widget/textinput.go

```diff
@@ -426,9 +434,14 @@ func (t *TextInput) drawTextAndCaret(screen *ebiten.Image, def DeferredRenderFun
 	tr := rect
 	tr = tr.Add(img.Point{t.padding.Left, t.padding.Top})
 
+	inputStr := t.InputText
+	if t.secure {
+		inputStr = strings.Repeat("*", len(t.InputText))
```

Comment:
This will not work correctly with Unicode code points that will be encoded into multiple UTF-8 bytes, for example the Euro symbol (U+20AC). The string needs to be converted to a []rune first before taking the len:

```suggestion
		inputStr = strings.Repeat("*", len([]rune(t.InputText)))
```

Reply:
Fixed! I'm not so great with character encoding. Good catch.